### PR TITLE
Companies get refac

### PR DESCRIPTION
**Pull request by Oscar-Rydh**

### CODE REVIEW

File: web/controllers/company_category_controller.ex [hunk 1]
```diff
@@ -4,7 +4,16 @@ defmodule Nexpo.CompanyCategoryController do
...
 
   def index(conn, _params) do
-    company_categories = Repo.all(CompanyCategory)
+        query  =
+        from( category in Nexpo.CompanyCategory,
+         distinct: true,
+         join: attribute in assoc(category, :attributes),
+         preload: :attributes
+         )
+    IO.puts("Running index")
```
Comment (JoelKlint) [inline web/controllers/company_category_controller.ex]: Ta bort log till consol

File: test/acceptance/categories_acceptance_test.exs [hunk 2]
```diff
@@ -5,15 +5,17 @@ defmodule Nexpo.CategoriesAcceptanceTest do
...
     conn = conn |> get("/api/categories")
 
     assert json_response(conn, 200)
     response = Poison.decode!(conn.resp_body)["data"]
     assert length(response) == length(company_categories)
     assert Enum.count(response) == 3
-    assert Enum.map(response, fn entry -> Map.keys(entry) |> Enum.map(&String.contains?(&1, "Generated Category")) end)
+    assert response == nil
+    assert Enum.map(response, fn entry -> String.contains?(entry.title, "Generated Category") end)
+
```
Comment (JoelKlint) [inline test/acceptance/categories_acceptance_test.exs]: Assertas response den som nil och icke-nil?

File: web/models/company_category.ex [hunk 3]
```diff
@@ -1,6 +1,8 @@
 defmodule Nexpo.CompanyCategory do
   use Nexpo.Web, :model
-
+   # This modell represents a company:
```
Comment (JoelKlint) [inline web/models/company_category.ex]: typo på model

File: test/acceptance/categories_acceptance_test.exs (hunk 2 above)
Comment (Oscar-Rydh) [inline test/acceptance/categories_acceptance_test.exs]: Nä, det är min variant av att debugga test

**Comment by Oscar-Rydh** [comment]
Håller med dina förslag! La upp så du kunde fortsätta om du ville idag :)

**Comment by Oscar-Rydh** [comment]
Fixar detta imorgon, kommer snickra om så att checken på om en child finns görs på samma sätt som du gjort för consistency

**Comment by JoelKlint** [comment]
Jag har tagit bort en del komplexa tester, och skrivit några nya simplare. Alla tester fungerar nu, men travis funkar inte korrekt förrän PR #41, därför failar React-testerna i denna PR.

Jag märkte att det svåraste är inte att skriva tester, eller att skapa funktionalitet. Det svåraste är att veta VAD vi ska testa efter. Att komma överens om hur API:t ska fungera. Vi borde sätta oss ner och bena ut alla detaljer för att underlätta det

**Comment by JoelKlint** [comment]
closes #22. Structure has been redone, which removes the problem from the first place

**Comment by JoelKlint** [comment]
Starts the work for #29. Before closing issue, we should make sure we have agreed on a structure and document it further

**Comment by Oscar-Rydh** [comment]
Har nyss lyckats sätta mig och kolla igenom, snyggt med view supporten! Dock om jag fatta din implementation rätt så vill du inte ha /attributes alls? Kan hålla med om det, get categories säger allt ändå

**Comment by JoelKlint** [comment]
Jag har faktiskt inte tänkt igenom om vi ska ha /attributes eller inte, men jag tyckte det var så många endpoints, så jag skalade ner de i antalet, och försökte fokusera på att få /companies och /categories implementerade. (De tycker jag är viktigast)

**Comment by Oscar-Rydh** [comment]
Håller med! Attributes är rätt lätt att aktivera om så skulle vara fallet senare. Annars tycker jag vi följer strukturen som vi definerade i /docs/entity-definitions.json

**Comment by JoelKlint** [comment]
Ja det låter bra. Det finns edge-cases dock. T.ex hur ser /companies ut om det inte finns några categories? etc.

**Comment by Oscar-Rydh** [comment]
True True. Ska vi lämna denna PR öppen, eller ska vi closa om den får travis test att gå igenom?

**Comment by JoelKlint** [comment]
Jag är sugen på att merga in i master så att testerna blir gröna.

Jag tycker merge vore najs också så backenden har grundlig funktionalitet. Då blir det lättare att kunna arbeta på frontenden också. Men jag mergar in master här först så testerna kan köras, om de går igenom tycker jag vi kör :)

**Comment by Oscar-Rydh** [comment]
Waayy, allt är grönt, så jag trycker in det!
